### ISSUE TITLE
Built output error 1.25.0-8.1.pre with enabled NullSafety

### ISSUE THREAD

**Issue by kaparray** [dart-lang/build#2962]
An error occurs when running the script: 

```
flutter packages pub run build_runner build
```
**Output:**
```
[INFO] Generating build script...
[INFO] Generating build script completed, took 346ms

[INFO] Creating build script snapshot......
[INFO] Creating build script snapshot... completed, took 6.7s

[SEVERE] Failed to snapshot build script .dart_tool/build/entrypoint/build.dart.
This is likely caused by a misconfigured builder definition.
[SEVERE] .dart_tool/build/entrypoint/build.dart:44:3: Warning: Operand of null-aware operation '?.' has type 'SendPort' which excludes null. - 'SendPort' is from 'dart:isolate'.  sendPort?.send(result);  ^Error: Cannot run with sound null safety, because the following dependenciesdon't support null safety: - package:build_runner_core - package:built_redux - package:built_value_generator - package:mockito - package:reflectable - package:build_config - package:source_gen - package:build_runner - package:build - package:analyzer - package:code_builder - package:dart_style - package:glob - package:crypto - package:logging - package:watcher - package:build_resolvers - package:json_annotation - package:timing - package:graphs - package:package_config - package:yaml - package:pub_semver - package:checked_yaml - package:pubspec_parse - package:build_daemon - package:args - package:io - package:convert - package:built_collection - package:built_value - package:quiver - package:_fe_analyzer_shared - package:http_multi_server - package:shelf - package:stream_transform - package:http_parser - package:mime - package:shelf_web_socket - package:web_socket_channel - package:cli_util - package:fixnumFor solutions, see https://dart.dev/go/unsound-null-safety.dart_tool/build/entrypoint/build.dart:42:44: Error: The parameter 'sendPort' can't have a value of 'null' because of its type 'SendPort', but the implicit default value is 'null'. - 'SendPort' is from 'dart:isolate'.Try adding either an explicit non-'null' default value or the 'required' modifier.void main(List<String> args, [_i8.SendPort sendPort]) async {                                           ^^^^^^^^
pub finished with exit code 78
```

### Flutter doctor:
```shell
flutter doctor -v
```
**Output:**
```
[✓] Flutter (Channel beta, 1.25.0-8.1.pre, on Mac OS X 10.15.7 19H2 darwin-x64, locale en-GB)
    • Flutter version 1.25.0-8.1.pre at /Users/kirill/development/flutter
    • Framework revision 8f89f6505b (5 days ago), 2020-12-15 15:07:52 -0800
    • Engine revision 92ae191c17
    • Dart version 2.12.0 (build 2.12.0-133.2.beta)

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/kirill/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.17.0

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • 8D2C5232-2C7D-4383-8193-86C2FF6E9ADF • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3
      (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 85.0.4183.121

! Doctor found issues in 1 category.

**Comment by simolus3** [comment]
Which version of `build_runner` are you using? Version `1.10.4` or later opts the script out of null-safety, which might help?

**Comment by jakemac53** [comment]
Yes this should be fixed by updating to version `1.10.4`
